### ISSUE TITLE
Views on remote files with DuckDB

### ISSUE THREAD

**Issue by soobrosa** [TobikoData/sqlmesh#2420]
This is my example `model.sql`

```
MODEL (
    name dwh.demo,
    kind VIEW
);

CALL load_aws_credentials();

SELECT
    column1,
    column2
FROM
    READ_PARQUET('s3://bucket/demo.parquet');
```

Running `sqlmesh plan dev` flies fine, view gets created.

`sqlmesh plan` gives
```
Error: Failed processing SnapshotId<"db"."dwh"."demo": 2291498964>. HTTP Error: HTTP GET error on 'https://bucket.s3.amazonaws.com/demo.parquet' (HTTP 403)
```

Sounds fair, not sure when should I do a pre `CALL load_aws_credentials();` Though I could supress this error with making a `schema.yaml`

```
- name: dwh.demo
  description: An external view
  columns:
    column1: text
    column2: text
```

What would be a ~canonical way of dealing with views on remote files? Only thing I can think of is making the views on the db before running SQLmesh then consider all of them as external tables. Neither feels right.

**Comment by izeigerman** [comment]
@soobrosa  What you've got should've worked assuming the authentication was successful. Any chance you can share the logs?

**Comment by soobrosa** [comment]
Thank you @izeigerman, sure!

```
Directly Modified: dwh.demo (Breaking)
Apply - Virtual Update [y/n]: y
Creating physical tables ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100.0% • 4/4 • 0:00:00

All model versions have been created successfully

2024-04-11 10:48:47,865 - MainThread - sqlglot - WARNING - 'CALL load_aws_credentials()' contains unsupported syntax. Falling back to parsing as a 'Command'. (parser.py:1350)
2024-04-11 10:48:47,866 - MainThread - sqlglot - WARNING - 'CALL load_aws_credentials()' contains unsupported syntax. Falling back to parsing as a 'Command'. (parser.py:1350)
Virtually Updating 'prod' ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0% • 0:00:00
2024-04-11 10:48:48,127 - MainThread - sqlmesh.core.context - ERROR - Apply Failure: Traceback (most recent call last):
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/utils/concurrency.py", line 227, in sequential_apply_to_dag
    fn(node)
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/utils/concurrency.py", line 165, in <lambda>
    lambda s_id: fn(snapshots_by_id[s_id]),
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/snapshot/evaluator.py", line 201, in <lambda>
    lambda s: self._promote_snapshot(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/snapshot/evaluator.py", line 693, in _promote_snapshot
    _evaluation_strategy(snapshot, self.adapter).promote(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/snapshot/evaluator.py", line 1042, in promote
    self.adapter.create_view(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/engine_adapter/shared.py", line 260, in internal_wrapper
    return func(*list_args, **kwargs)
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/engine_adapter/base.py", line 873, in create_view
    self.execute(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/engine_adapter/base.py", line 1772, in execute
    self._execute(sql, **kwargs)
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/engine_adapter/base.py", line 1778, in _execute
    self.cursor.execute(sql, **kwargs)
duckdb.duckdb.HTTPException: HTTP Error: HTTP GET error on 's3://bucket/demo.parquet' (HTTP 403)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/context.py", line 1089, in apply
    self._scheduler.create_plan_evaluator(self).evaluate(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/plan/evaluator.py", line 119, in evaluate
    self._update_views(plan, promotion_result, deployability_index_for_evaluation)
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/plan/evaluator.py", line 257, in _update_views
    self.snapshot_evaluator.promote(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/core/snapshot/evaluator.py", line 199, in promote
    concurrent_apply_to_snapshots(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/utils/concurrency.py", line 163, in concurrent_apply_to_snapshots
    return concurrent_apply_to_dag(
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/utils/concurrency.py", line 198, in concurrent_apply_to_dag
    return sequential_apply_to_dag(dag, fn, raise_on_error)
  File "/Users/soobrosa/github/dwh/.venv/lib/python3.10/site-packages/sqlmesh/utils/concurrency.py", line 230, in sequential_apply_to_dag
    raise NodeExecutionFailedError(node) from ex
sqlmesh.utils.concurrency.NodeExecutionFailedError: Execution failed for node SnapshotId<"db"."dwh"."demo": 2291498964>
 (context.py:1096)
Error: Failed processing SnapshotId<"db"."dwh"."demo": 2291498964>. HTTP Error: HTTP GET error on 's3://bucket/demo.parquet' (HTTP 403)
```

**Comment by soobrosa** [comment]
Adding also the related log.
[sqlmesh_2024_04_11_10_53_21.log](https://github.com/TobikoData/sqlmesh/files/14946123/sqlmesh_2024_04_11_10_53_21.log)

**Comment by izeigerman** [comment]
It looks like the failure occurs when we try to create the view in the Virtual Layer, not the model view itself:
```
2024-04-11 10:53:26,040 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Updating view 'db.dwh.demo' to point at table 'db.sqlmesh__dwh.dwh__demo__2346974520' (evaluator.py:1041)
2024-04-11 10:53:26,040 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: CREATE OR REPLACE VIEW "db"."dwh"."demo" AS SELECT * FROM "db"."sqlmesh__dwh"."dwh__demo__2346974520" (base.py:1775)
```
For whatever reason, duckdb needs to access the URL as a result of a simple view creation. This behavior doesn't make much sense to me. I'll have to think how to address this.

**Comment by soobrosa** [comment]
Thank you @izeigerman! Maybe from the user perspective it looks like adding `authentication` to the `duckdb` engine?

**Comment by soobrosa** [comment]
Solved with

```
CREATE PERSISTENT SECRET my_persistent_secret (
    TYPE S3,
    PROVIDER CREDENTIAL_CHAIN,
    ENDPOINT 's3.eu-west-2.amazonaws.com'
);
```

(you have to have valid `~/.aws/credentials`, change your ENDPOINT accordingly) and running

```
chmod 600 ~/.duckdb/stored_secrets/my_persistent_secret.duckdb_secret
```
afterwards.

No need for

```
CALL load_aws_credentials();
```

**Comment by treysp** [comment]
Link to DuckDB secrets manager (available >=v0.10.0) https://duckdb.org/docs/configuration/secrets_manager.html

**Comment by soobrosa** [comment]
Reason for 600 https://github.com/duckdb/duckdb/blob/eb9f63a1cd506afe1fad8125d6f123868e979b5d/src/include/duckdb/common/local_file_system.hpp#L96
